src/Layout.c

```diff
@@ -770,9 +777,112 @@ static void layoutNodeImpl(css_node_t *node, float parentMaxWidth) {
 
     linesCrossDim += crossDim;
     linesMainDim = fmaxf(linesMainDim, mainDim);
+    linesCount += 1;
     startLine = endLine;
   }
 
+  // <Loop DD>
+  //
+  // PIERRE: More than one line, we need to layout the crossAxis according to
+  // alignContent.
+  //
+  // Note that we could probably remove <Loop D> and handle the one line case
+  // here too, but for the moment this is safer since it won't interfere with
+  // previously working code.
+  //
+  // See specs:
+  // http://www.w3.org/TR/2012/CR-css3-flexbox-20120918/#layout-algorithm
+  // section 9.4
+  //
+  if (linesCount > 1 &&
+      (!isUndefined(node->layout.dimensions[dim[crossAxis]])))
+  {
+    float nodeCrossAxisInnerSize = node->layout.dimensions[dim[crossAxis]] -
+        getPaddingAndBorderAxis(node, crossAxis);
+    float remainingCrossDim = nodeCrossAxisInnerSize - linesCrossDim;
+
+    float crossDimAdd = 0;
+    float currentLead = getPaddingAndBorder(node, leading[crossAxis]);
+
+    css_align_t alignContent = getAlignContent(node);
+    if (alignContent == CSS_ALIGN_FLEX_END) {
+      currentLead += remainingCrossDim;
+    }
+    else if (alignContent == CSS_ALIGN_CENTER) {
+      currentLead += remainingCrossDim / 2;
+    }
+    else if (alignContent == CSS_ALIGN_STRETCH) {
+      if (nodeCrossAxisInnerSize > linesCrossDim) {
+        crossDimAdd = (remainingCrossDim / linesCount);
+      }
+    }
+
+    // find the first node on the first line
+    for (i = 0; i < node->children_count; ) {
+      int startIndex = i;
+      int lineIndex = -1;
+
+      // get the first child on the current line
+      {
```

Comment:
I was declaring another variable in there, I switched to use the child variable when I converted the code to JS/transpiler. Removed it.